cli/tsc.rs

```diff
@@ -63,6 +64,44 @@ fn get_maybe_hash(
   }
 }
 
+/// tsc only supports `.ts`, `.tsx`, `.d.ts`, `.js`, or `.jsx` as root modules
+/// and so we have to detect the apparent media type based on extensions it
+/// supports.
+fn get_tsc_media_type(specifier: &ModuleSpecifier) -> MediaType {
+  let url = specifier.as_url();
+  let path = if url.scheme() == "file" {
+    if let Ok(path) = url.to_file_path() {
+      path
+    } else {
+      PathBuf::from(url.path())
+    }
+  } else {
+    PathBuf::from(url.path())
+  };
+  match path.extension() {
+    None => MediaType::Unknown,
+    Some(os_str) => match os_str.to_str() {
+      Some("ts") => match path.file_stem() {
+        Some(os_str) => match os_str.to_str() {
+          Some(file_name) => {
+            if file_name.ends_with(".d") {
+              MediaType::Dts
+            } else {
+              MediaType::TypeScript
+            }
+          }
+          None => MediaType::TypeScript,
+        },
```

Comment:
nit: this `match path.file_stem()` could maybe be written more cleanly using `if let Some(...) `